server/views/pages/BCST2-form.njk

```diff
@@ -32,21 +34,34 @@
         {% if not assessmentPage.error %}
           <h1 class="govuk-heading-l govuk-!-margin-top-6 govuk-!-margin-bottom-3">{{ assessmentPage.title }}</h1>
           <form action="/BCST2-next-page/?prisonerNumber={{ prisonerData.personalDetails.prisonerNumber }}" method="post">
-          
-            {% for questionAndAnswer in assessmentPage.questionsAndAnswers %}
-              {% if questionAndAnswer.question.type === 'RADIO_WITH_ADDRESS' %}
-                {{ radioWithAddress(questionAndAnswer) }}
+            
+            {% if assessmentPage.id === "ASSESSMENT_SUMMARY" %}
+              {{ assessmentSummary(pathway) }}
+            {% endif %}
+
+            {% if assessmentPage.id === "CHECK_ANSWERS" %}
+              {{ checkAnswers(allQuestionsAndAnswers, pathway, prisonerData.personalDetails.prisonerNumber) }}
+            {% endif %}
+
+            {% for currentQuestionAndAnswer in assessmentPage.questionsAndAnswers %}
+
+              {# get answer to current question #}
+              {% set existingAnswer = currentQuestionAndAnswer | getAnswerToCurrentQuestion(allQuestionsAndAnswers) %}
+
+              {% if currentQuestionAndAnswer.question.type === 'RADIO_WITH_ADDRESS' %}
+                {{ radioWithAddress(currentQuestionAndAnswer, existingAnswer) }}
               {% endif %}
-              {% if questionAndAnswer.question.type === 'RADIO' %}
-                {{ radio(questionAndAnswer) }}
+              {% if currentQuestionAndAnswer.question.type === 'RADIO' %}
+                {{ radio(currentQuestionAndAnswer, existingAnswer) }}
               {% endif %}
-              {% if questionAndAnswer.question.type === 'LONG_TEXT' %}
-                {{ longText(questionAndAnswer) }}
+              {% if currentQuestionAndAnswer.question.type === 'LONG_TEXT' %}
+                {{ longText(currentQuestionAndAnswer, existingAnswer) }}
               {% endif %}
-              {% if questionAndAnswer.question.type === 'DROPDOWN' %}
-                {{ dropDown(questionAndAnswer) }}
+              {% if currentQuestionAndAnswer.question.type === 'DROPDOWN' %}
+                {{ dropDown(currentQuestionAndAnswer, existingAnswer) }}
               {% endif %}
```

Comment:
Might be worth changing these to be a series of `if else`s or a switch statement (if njk has that) so we can put in an else condition if the question type doesn't match any of the expected types and display an error.